operator/autoscaler.go

```diff
@@ -283,12 +300,22 @@ func (as *AutoScaler) scaleUpOrDown(esIndices map[string]ESIndex, scalingHint Sc
 				})
 			}
 			if newTotalShards != currentTotalShards {
-				newDesiredNodeReplicas := as.ensureUpperBoundNodeReplicas(scalingSpec, calculateNodesWithSameShardToNodeRatio(currentDesiredNodeReplicas, currentTotalShards, newTotalShards))
+				newDesiredNodeReplicas := currentDesiredNodeReplicas
+
+				currentShardToNodeRatio = float64(newTotalShards) / float64(currentDesiredNodeReplicas)
```

Comment:
what you're calculating here is not the current shard-to-node ratio, it is actually the target shard-to-node ratio if we kept the same number of nodes. Therefore the log message below "Keeping shard-to-node ratio (%.2f), and increasing index replicas." would be wrong.

Reply:
yes, it's a good catch. I will fix it

Reply:
👍